onnxruntime/core/session/inference_session.cc

```diff
@@ -349,6 +349,14 @@ common::Status InferenceSession::RegisterGraphTransformer(
   if (p_graph_transformer == nullptr) {
     return Status(common::ONNXRUNTIME, common::FAIL, "Received nullptr for graph transformer");
   }
+
+  if (is_inited_) {
```

Comment:
Should this be guarded by session_mutex_?

Reply:
Don't think it matters here but can call IsInitialized instead.

---
In reply to: [443771207](https://github.com/microsoft/onnxruntime/pull/4194#discussion_r443771207) [](ancestors = 443771207)